test: change structure of fixtures for `blas/base/dtrmv`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

none

## Description

> What is the purpose of this pull request?

This pull request:

-   Change the structure of fixtures so that it matches the order of parameters in the implementation and add `A_mat` for user reference matrix

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

hrshya committed:
bench: rename `randu` to `uniform` for consistency

PR-URL: https://github.com/stdlib-js/stdlib/pull/6043
Reviewed-by: Athan Reines <kgryte@gmail.com>

ShabiShett07 committed:
test: change structure of fixtures for blas/base/dtrmv

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/blas/base/dtrmv/test/fixtures/column_major_l_nt_nu.json:
```diff
@@ -1,16 +1,21 @@
+  "N": 3,
+  "A": [ 1.0, 2.0, 3.0, 0.0, 4.0, 5.0, 0.0, 0.0, 6.0 ],
+  "A_mat": [
+    [ 1.0, 2.0, 3.0 ],
```
@ShabiShett07 Correct me if I am wrong, but shouldn't `A_mat` be lower triangular per L3?

ShabiShett07 replied:
The reason why `A_mat` is in an upper triangular matrix, it us column major, so the elements are accesses in column-wise that means it should be written the form of `row-major upper` that was my thought related to this. Also you can see that `A`is written in the same way here.

kgryte replied:
Whether it is stored in row-major or column-major is not material, as `A_mat` should be from the perspective of the end-user. An array order is a storage layout, which should not be apparent to users. That means, if `A` is supposed to be lower triangular, `A_mat` should be lower triangular.

kgryte replied:
If `A` is column-major and not transposed, that means that data is stored column-wise. In which case,

```
A_mat = [
  [ 1.0, 0.0, 0.0 ],
  [ 2.0, 4.0, 0.0 ],
  [ 3.0, 5.0, 6.0 ]
]
```

ShabiShett07 replied:
Sure I will make the necessary changes

kgryte commented on lib/node_modules/@stdlib/blas/base/dtrmv/test/fixtures/column_major_complex_access_pattern.json:
```diff
@@ -1,14 +1,21 @@
+  "N": 3,
+  "A": [ 6.0, 999.0, 0.0, 999.0, 0.0, 5.0, 999.0, 4.0, 999.0, 0.0, 3.0, 999.0, 2.0, 999.0, 1.0 ],
+  "A_mat": [
+    [ 1.0, 2.0, 3.0 ],
```
Same comment. Lower triangular?

kgryte commented on lib/node_modules/@stdlib/blas/base/dtrmv/test/fixtures/column_major_l_t_nu.json:
```diff
@@ -1,16 +1,21 @@
+  "N": 3,
+  "A": [ 1.0, 2.0, 3.0, 0.0, 4.0, 5.0, 0.0, 0.0, 6.0 ],
+  "A_mat": [
+    [ 1.0, 2.0, 3.0 ],
```
@ShabiShett07 I suggest revisiting these and ensuring that `A_mat` correctly reflects `uplo`.

kgryte reviewed (changes requested):
Left initial comments.

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update fixtures

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM. Agreed that this makes the fixtures easier to read and follow. Thank you for doing this, @ShabiShett07!
